### PR TITLE
fix tx cache, storage

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/yoroi-wallets/storage.ts [hunk 1]
```diff
@@ -8,19 +8,23 @@ export type Storage = {
...
+      const prefixedKeys = keys.map((key) => withPrefix(key))
+      const items = await AsyncStorage.multiGet(prefixedKeys)
+
+      return items.map(([key, value]) => [withoutPrefix(key), value])
     },
     setItem: (key: string, value: string) => {
       return AsyncStorage.setItem(withPrefix(key), value)
     },
-    multiSet: (items: Array<[string, string]>) => {
+    multiSet: (items: Array<[key: string, value: string]>) => {
```
Comment (wolverineks) [inline src/yoroi-wallets/storage.ts:27]: prefer named tuples

File: src/yoroi-wallets/storage.ts [hunk 2]
```diff
@@ -8,19 +8,23 @@ export type Storage = {
...
+  const withPrefix = (key: string) => `${prefix}${key}`
+  const withoutPrefix = (value: string) => value.slice(prefix.length)
 
   return {
     getItem: (key: string) => {
       return AsyncStorage.getItem(withPrefix(key))
     },
-    multiGet: (keys: Array<string>) => {
-      return AsyncStorage.multiGet(keys.map((key) => withPrefix(key)))
+    multiGet: async (keys: Array<string>) => {
```
Comment (wolverineks) [inline src/yoroi-wallets/storage.ts:18]: remove prefix, this was the bug

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 3]
```diff
@@ -508,10 +508,38 @@ const parseTxids = (data: string | null | undefined) => {
   return isTxids(txids) ? txids : []
 }
 
+const parseTx = (data: string | null | undefined): Transaction | undefined => {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts:511]: better parsing, just cleaning up

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 4]
```diff
@@ -471,29 +471,29 @@ const makeTxCacheStorage = (storage: Storage): TxCacheStorage => ({
...
+
+    const tuples = await storage.multiGet(txids)
 
-    const values = await storage.multiGet(txids)
-    return values.reduce((result, [txid, tx]) => {
-      if (!tx) return result
-      try {
-        return {...result, [txid]: JSON.parse(tx)}
-      } catch (_error) {
+    return tuples.reduce((result: TransactionCacheState['transactions'], [txid, txData]) => {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts:478]: tuples was originally `[path: string, transaction: string]`
now its `[txid: string, transaction: string]`

File: src/yoroi-wallets/cardano/shelley/transactionCache.test.ts [hunk 5]
```diff
@@ -79,9 +79,12 @@ const mockStorage: Storage = {
     if (path === 'txids') return JSON.stringify([mockTx.id])
     throw new Error('invalid path')
   },
-  multiGet: async (paths: Array<string>) => {
-    if (paths.length === 1 && paths[0] === mockTx.id) return [[mockTx.id, JSON.stringify(mockTx)]]
-    throw new Error('invalid path')
+  multiGet: async (txids: Array<string>) => {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.test.ts:82]: clean up

File: src/yoroi-wallets/cardano/ShelleyWallet.ts [hunk 6]
```diff
@@ -257,7 +257,7 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
     this.rewardAddressHex = await deriveRewardAddressHex(this.publicKeyHex, this.networkId)
     this.isEasyConfirmationEnabled = data.isEasyConfirmationEnabled
 
-    this.transactionCache = await TransactionCache.create(makeStorageWithPrefix(`/wallet/${walletMeta.id}/txs`))
+    this.transactionCache = await TransactionCache.create(makeStorageWithPrefix(`/wallet/${walletMeta.id}/txs/`))
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/ShelleyWallet.ts:260]: moved `"/"` from `makeStorageWithPrefix` to make removing prefix simpler

**Review by banklesss — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

**Pull request merged**